src/java.base/share/classes/java/lang/invoke/MemberName.java

```diff
@@ -314,20 +314,22 @@ private boolean isObjectPublicMethod() {
     boolean referenceKindIsConsistentWith(int originalRefKind) {
         int refKind = getReferenceKind();
         if (refKind == originalRefKind)  return true;
-        switch (originalRefKind) {
-        case REF_invokeInterface:
-            // Looking up an interface method, can get (e.g.) Object.hashCode
-            assert(refKind == REF_invokeVirtual ||
-                   refKind == REF_invokeSpecial) : this;
-            return true;
-        case REF_invokeVirtual:
-        case REF_newInvokeSpecial:
-            // Looked up a virtual, can get (e.g.) final String.hashCode.
-            assert(refKind == REF_invokeSpecial) : this;
-            return true;
-        }
-        assert(false) : this+" != "+MethodHandleNatives.refKindName((byte)originalRefKind);
-        return true;
+        return switch (originalRefKind) {
+            case REF_invokeInterface -> {
+                // Looking up an interface method, can get (e.g.) Object.hashCode
+                assert (refKind == REF_invokeVirtual || refKind == REF_invokeSpecial) : this;
+                yield true;
+            }
+            case REF_invokeVirtual, REF_newInvokeSpecial -> {
+                // Looked up a virtual, can get (e.g.) final String.hashCode.
+                assert (refKind == REF_invokeSpecial) : this;
+                yield true;
+            }
+            default -> {
+                assert(false) : this+" != "+MethodHandleNatives.refKindName((byte)originalRefKind);
+                yield true;
+            }
```

Comment:
this code always yield true, better to check if the assert are enabled, do the switch in that case and always return true

Reply:
Thanks for your suggestion. I've incorporated it into commit a8706b0